cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -327,6 +327,30 @@ private synchronized void removeOldUserTasks() {
     }
   }
 
+  /**
+   * Allow other subsystems to retrieve {@link UserTaskInfo} using uuid. This method assumes uuid comes from some internal book
+   * keeping state, not an external http request. Thus it does not check request parameters like
+   * {@link UserTaskManager#getUserTaskByUserTaskId} below does.
+   * @param uuid UUID associated with a user request or execution.
+   * @return {@link UserTaskInfo} which contains information about a user request.
+   */
+  public synchronized UserTaskInfo getUserTaskById(String uuid) {
+    if (uuid == null) {
+      return null;
+    }
+
+    UUID userTaskId = UUID.fromString(uuid);
+    if (_allUuidToUserTaskInfoMap.get(TaskState.COMPLETED).containsKey(userTaskId)) {
+      return _allUuidToUserTaskInfoMap.get(TaskState.COMPLETED).get(userTaskId);
+    }
+
+    if (_allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).containsKey(userTaskId)) {
+      return _allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).get(userTaskId);
```

Comment:
What does `return _allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).get(userTaskId);` return if `userTaskId` does not exist in `_allUuidToUserTaskInfoMap.get(TaskState.ACTIVE)` map? 

Would it possible to improve our implementation using this information?

Reply:
Do you mean before when `Executor` tries to get `_userTaskInfo` but the `_userTaskInfo` has not been inserted into `_allUuidToUserTaskInfoMap.get(TaskState.ACTIVE)`? Or do you mean `userTaskId` isn't found in  `_allUuidToUserTaskInfoMap`?